tests/executor.rs

```diff
@@ -0,0 +1,73 @@
+extern crate jni;
+
+use jni::Executor;
+use jni::sys::jint;
+
+pub mod util;
+use util::{jvm, AtomicIntegerProxy};
+use std::thread::spawn;
+use std::sync::{Arc, Barrier};
+
+#[test]
+fn single_thread() {
+    let executor = Executor::new(jvm().clone());
+    test_single_thread(executor);
+}
+
+#[test]
+fn serialized_threads() {
+    let executor = Executor::new(jvm().clone());
+    test_serialized_threads(executor);
+}
+
+#[test]
+fn concurrent_threads() {
+    let executor = Executor::new(jvm().clone());
+    const THREAD_NUM: usize = 8;
+    test_concurrent_threads(executor, THREAD_NUM)
+}
+
+fn test_single_thread(executor: Executor) {
+    let mut atomic = AtomicIntegerProxy::new(executor, 0).unwrap();
+    assert_eq!(0, atomic.get().unwrap());
+    assert_eq!(1, atomic.increment_and_get().unwrap());
+    assert_eq!(3, atomic.add_and_get(2).unwrap());
+    assert_eq!(3, atomic.get().unwrap());
+}
+
+fn test_serialized_threads(executor: Executor) {
+    let mut atomic = AtomicIntegerProxy::new(executor, 0).unwrap();
+    assert_eq!(0, atomic.get().unwrap());
+    let jh = spawn(move || {
+        assert_eq!(1, atomic.increment_and_get().unwrap());
+        assert_eq!(3, atomic.add_and_get(2).unwrap());
+        atomic
+    });
+    let mut atomic = jh.join().unwrap();
+    assert_eq!(3, atomic.get().unwrap());
+}
+
+fn test_concurrent_threads(executor: Executor, thread_num: usize) {
+    const ITERS_PER_THREAD: usize = 10_000;
```

Comment:
How long does this test take? :upside_down_face:

Reply:
About one second, I think it's affordable.